### PR TITLE
New sciline release

### PR DESCRIPTION
**Pull request by jokasimr**
Prepares the package for the new sciline release.

Not sure if there are more things that should be done. Please take a look an make some suggestions.

The behavior changes slightly in this PR despite it being a refactoring, because there is still no mechanism for optional inputs in providers as far as I understand, and the orso file writing relied on the old behavior there.

### CODE REVIEW

File: docs/examples/amor.ipynb [hunk 1]
```diff
@@ -129,19 +129,34 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "pl[TutorialFilename[Sample]] = \"amor2023n000608.hdf\"\n",
```
Comment (nvaytet) [inline docs/examples/amor.ipynb:132]: After discussion on Slack: maybe this is a case where traditional looping would be less painful?
We are already caching the computation of the reference manually, so we could do
```Py
runs = {
    "amor2023n000608.hdf": sc.scalar(0.85, unit='deg'),
    "amor2023n000609.hdf": sc.scalar(2.25, unit='deg'),
    "amor2023n000610.hdf": sc.scalar(3.65, unit='deg'),
}

results = {}
for file, angle in runs.items():
    pl[TutorialFilename[Sample]] = file
    pl[SampleRotation[Sample]] = angle
    results[file] = pl.compute(NormalizedIofQ).hist()

sc.plot(results, norm='log', vmin=1e-4)
```
?

Unless using cyclebane would mean that we could remove the manual reference caching step?

Comment (jokasimr) [reply docs/examples/amor.ipynb:132]: In practice we will read the reference from a file anyway, so it doesn't make much sense to have the reference reduction be part of the pipeline that reduces samples.

I'm okay with doing it the 'traditional way' here.

**Comment by nvaytet** [comment]
The changes here are small, so I think it doesn't matter if this or #51 gets merged first.

**Comment by jokasimr** [comment]
> The changes here are small, so I think it doesn't matter if this or https://github.com/scipp/essreflectometry/pull/51 gets merged first.

That's fine with me, go ahead and merge #51.

File: pyproject.toml [hunk 2]
```diff
@@ -36,7 +36,7 @@ dependencies = [
   "plopp",
   "pythreejs",
   "orsopy",
-  "sciline>=23.9.1",
+  "sciline @ git+https://github.com/scipp/sciline@main",
```
Comment (nvaytet) [inline pyproject.toml:39]: I think the `sciline` release has been made now, so we should go back to not using `main` and re-run the `tox -e deps`?

**Review by nvaytet — APPROVED** [review]

**Pull request merged**